site/en/codelabs/esp-openthread-hardware/index.lab.md

```diff
@@ -0,0 +1,600 @@
+I(69089) OPENTHREAD:[N] Mle-----------: Role detached -> leader
+I(69089) OPENTHREAD:[N] Mle-----------: Partition ID 0x28b518c6
+I (69099) OPENTHREAD: Platform UDP bound to port 49154
+```
+
+After a moment, check the device state. It should be the Leader.
+
+```console
+> state
+
+leader
+Done
+> 
+```
+
+## Join the thread network via networkkey
+
+Duration: 3:00
+
+In this codelab, two ESP boards with IEEE 802.15.4 module are prepared for joining the BR formed network. In this session, we will add Board1 to the network.
+
+Get the networkkey from BR:
+
+```console
+
+## BR Commissioner ##
+----------------------
+> networkkey
+
+731ab6a60a64a0a0b14b259b86b2be01
+Done
+> 
+```
+
+Set this networkkey to one ESP board(Board1 Joiner) with IEEE 802.15.4 module:
+
+```console
+## Board1 Joiner ##
+----------------------
+> dataset networkkey 731ab6a60a64a0a0b14b259b86b2be01
+
+Done
+```
+
+Commit this dataset as the active one:
+
+```console
+> dataset commit active
+
+Done
+```
+
+Bring up the IPv6 interface:
+
+```console
+> ifconfig up
+
+Done
+I (20308) OT_STATE: netif up
+```
+
+Start Thread protocol operation:
+
+```console
+> thread start
+
+I(23058) OPENTHREAD:[N] Mle-----------: Role disabled -> detached
+Done
+> I(23408) OPENTHREAD:[N] Mle-----------: Attach attempt 1, AnyPartition reattaching with Active Dataset
+I(30028) OPENTHREAD:[N] Mle-----------: Attach attempt 1 unsuccessful, will try again in 0.288 seconds
+I(30328) OPENTHREAD:[N] Mle-----------: Attach attempt 2, AnyPartition 
+I(33498) OPENTHREAD:[N] Mle-----------: Delay processing Announce - channel 21, panid 0x1444
+I(33758) OPENTHREAD:[N] Mle-----------: Processing Announce - channel 21, panid 0x1444
+I(33758) OPENTHREAD:[N] Mle-----------: Role detached -> disabled
+I(33758) OPENTHREAD:[N] Mle-----------: Role disabled -> detached
+I(34178) OPENTHREAD:[N] Mle-----------: Attach attempt 1, AnyPartition 
+I(35068) OPENTHREAD:[N] Mle-----------: RLOC16 fffe -> 2c01
+I(35068) OPENTHREAD:[N] Mle-----------: Role detached -> child
+```
+
+After a moment, check the device state. It should be the Child.
+
+```console
+> state
+
+child
+Done
+```
+
+Set the role to router.
+
+```console
+> state router
+
+Done
+I(51028) OPENTHREAD:[N] Mle-----------: RLOC16 2c01 -> 2800
+I(51028) OPENTHREAD:[N] Mle-----------: Role child -> router
+I(51028) OPENTHREAD:[N] Mle-----------: Partition ID 0x28b518c6
+> 
+```
+
+<img src="img/esp_ot_tp_LR.jpg" alt="esp_ot_tp_LR.jpg"  width="400"/>
+
+## Join the thread network via secure commissioning
+
+Duration: 5:00
+
+In this session, we will add Board2 to the network via security commissioning:
+
+Get the PSKc and panid from BR Commissioner:
+
+```console
+
+## BR Commissioner ##
+----------------------
+> pskc
+
+54e7f18d2575014da94db09df29c5df0
+Done
+> panid
+
+0x1444
+Done
+```
+
+Configure the network information to Board2:
+
+
+```console
+
+## Board2 Joiner ##
+----------------------
+
+> dataset pskc 54e7f18d2575014da94db09df29c5df0
```

Comment:
Jeff B says: "For consistency across examples don't have a blank line between the ------- and the command/output"